sdk/ml/azure-ai-ml/azure/ai/ml/operations/_feature_store_operations.py

```diff
@@ -162,9 +163,9 @@ def get(self, name: str, **kwargs: Dict) -> FeatureStore:
                     )
 
             # materialization identity = identity when created through feature store operations
-            if (offline_store_connection and offline_store_connection.name == OFFLINE_STORE_CONNECTION_NAME) or (
-                online_store_connection and online_store_connection.name == ONLINE_STORE_CONNECTION_NAME
-            ):
+            if (
+                offline_store_connection and offline_store_connection.name.startswith(OFFLINE_STORE_CONNECTION_NAME)
+            ) or (online_store_connection and online_store_connection.name.startswith(ONLINE_STORE_CONNECTION_NAME)):
```

Comment:
We can also remove the reference in FeatureStoreSettings. That default value could cause confusion in future.

https://github.com/anyalee0221/azure-sdk-for-python/blob/8ec60929e5bc8fb0f23591048fd26785b8e5a4dc/sdk/ml/azure-ai-ml/azure/ai/ml/entities/_feature_store/feature_store.py#L26

Reply:
Removed it, changed to set the name when call begin_create instead.